ugs-platform/ugs-platform-surfacescanner/src/main/java/com/willwinder/ugs/platform/surfacescanner/AutoLevelerTopComponent.java

```diff
@@ -599,26 +599,32 @@ private void dataViewerActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FI
     }//GEN-LAST:event_dataViewerActionPerformed
 
     private void applyToGcodeActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_applyToGcodeActionPerformed
-        GcodeParser gcp = new GcodeParser();
```

Comment:
Was something broken with this approach? Without replacing the gcode parser you could have other processors loaded which might effect the final generated program.

Reply:
I noticed that `applyGcodeParser` was deprecated and I assumed that having all processors that loaded run would be the expected behavior...wouldn't one expect those other processors to effect the final generated program?

Reply:
I remember deprecating this method as I thought it got unintuitive when you applied a processor the previous one got replaced. For instance this prevented the user to apply "Mirror" and "Run from" at the same time.

It is still a bit unintuitive as the changes aren't reflected in the GCode which might have made more sense.

Reply:
+1 to reflecting the changes in the GCode in some way. I considered adding a (read-only) view of what will actually be sent out when I was debugging this.

Reply:
The default processors would remove comments and I always tried to maintain those in the editor view, but I can see how it's misleading to display different gcode than what gets sent.

Reply:
Would it not be best to apply _all_ processors in a serial fashion, rather than just the last? The order could be pre-determined based on what made the most sense.